### ISSUE TITLE
Проблемы с некоторыми либами

### ISSUE THREAD

**Issue by krashaen** [fullstack-development/react-redux-starter-kit#116]
Возможно тут я сейчас не прав и вы меня поправите, либо поправим эти моменты)
1) `doiuse`, тут мне кажется нужно настройки для браузеров и игнорирования определенных свойств нужно вынести в отдельный файлик, я с ходу не разобрался сразу как заигнорить определенное свойство)
2) Почему убрали `inline-svg`? Тут я может чуток отстал, но у ребят спрашивал, вроде все им пользуются. Есть другой способ вставки svg в дом? 
3) `postcssReporter` - как то странно, он то подсвечивает ошибки, то нет, например ставлю свойство `resize:none`, когда делал он не показывал на что именно ругается ошибка была только такая:
![image](https://user-images.githubusercontent.com/12808495/59547360-e6e9db80-8f67-11e9-82eb-ec01740818fd.png)

Но на момент написания появилось сообщение

![image](https://user-images.githubusercontent.com/12808495/59547383-30d2c180-8f68-11e9-9a0c-e3601c3238a5.png)

может я просмотрел, но Дима Юдин и Саша Петухов тоже смотрели)

**Comment by Znack** [comment]
@in19farkt @krashaen @chmnkh как думаете, этим стоит заняться, если у кого-нибудь из наших разработчиков время будет?

**Comment by chmnkh** [comment]
я не эксперт в вопросах в сборки, мне вообще в целом не понятно, когда, кем, и при каких условиях всё это настраивалось

сейчас там куча всего и довольно всё запутано

возможно там вообще стоит всё пересмотреть, но это к Диме больше

**Comment by chmnkh** [comment]
стартер кит вообще великого рефакторинга требует, поговаривают, поэтому это лишь частная небольшая проблема
мне кажется если по уму стартеркит переписывать, то там и глобально по сборке надо глянуть, что происходит

**Comment by in19farkt** [comment]
Про сборку, я бы вообще какой-нибудь create-react-app посмотрел, там должно быть уже всё настроено в лучшем виде и вроде доступна кастомизация вебпак конфига

**Comment by Znack** [comment]
В create-react-app довольно жопная конфигурация конфигов, по крайней мере около года назад была, можешь посмотреть как из-за этого сделали create-react-app-rewired даже :)

**Comment by Znack** [comment]
я предлагаю сейчас в любом случае почистить сборку, скорее всего великий рефатокринг её не заденет, а то мы так вечно будем все задачи откладывать в ожидании чуда :) Даже если перепишем потом starter-kit этот, то текущие усилия вряд ли займут кучу времени

**Comment by clicktronix** [comment]
nextjs по-моему сейчас куда круче create react app кастомизируется, нет? хотя это прям фреймворк уже

**Comment by Znack** [comment]
Это тот, который не даёт нам самим папки организовывать, а связывает структуру и рауты? 😄

**Comment by in19farkt** [comment]
Да, там всё круто, но вот раутинг ты файлами делаешь. Но вроде бы это дает больше плюсов чем минусов, там сборка заточена под это, каждая страница тянет за собой только свои зависимости, а линка на другую страницу делает префетч скриптов другой страницы.

В общем для большинства проектов я думаю это вообще не проблема, а просто дело привычки. Если совсем приспичило, то можно попробовать на верхнем уровне побить на страницы так как хочет/умеет nextjs, а внутри каждой страницы для отрисовки использовать Route из react-router, для редиректов использовать линки из nextjs.

**Comment by Znack** [comment]
Ну мне просто кажется, что когда проект из стадии MVP будет переходить в стадию зрелого продукта, либо когда для MVP потребуется что-то кастомное, то станет очень больно. И я не совсем понимаю, какую проблему nextjs решает? Если только SSR, то там же всего файликом в 100-200 строк можно обойтись, чтобы себе SSR поднять, дейтствительно ли стоит из-за этого целый фреймворк подключать в проект?

**Comment by in19farkt** [comment]
- быстрая сборка/пересборка с нормальным хотрелоадом, кешированием и прочими ништяками
- рендеринг страниц на этапе сборки (возможно с префетчингом каких-то данных), полезно для лендинг страниц
- рендеринг страниц на сервере (тоже с префетчингом если нужно), нормально настроенный SSR (тут по-моему ты не дооцениваешь немного масштаб проблемы)
- на сколько помню (могу ошибаться, нужно перепроверить), автоматически бьет код по чанкам, каждая страница собирается отдельно и тянет за собой только свои зависимости, есть свои линки (вместо реатк-роутер линков), которые позволяют автоматически префетчить другие страницы. Тут возможно становятся не нужны наши асинхронные фичи, т.к. страницы и так норм оптимизированы из коробки
- в общем всё то что мы пытались сделать своими руками в своем стартерките и не преуспели в этом.
- вокруг этого большое комьюнити, которое уже делало на нексте всё что угодно, огромное кол-во примеров как решить ту или иную проблему

Это то что я вспомнил на ходу не углубляясь. Про проблемы на стадии зрелого продукта немного не понял, по сути этот фреймворк дает хорошую отправную точку, но при этом сильно тебя не ограничивает, вебпак конфиг можно расширять (причем довольно просто и удобно), по структуре проекта только страницы привязаны к раутам, в остальном делай как хочешь. Рауты-страницы вроде ни как не должны мешать, если нужно верхнеуровнево побили на страницы с помощью некста, а внутри страницы уже можем раутинг с помощью реакт роутера разрулить. Дока не большая, можно за полдня понять всё что нужно, чтобы начать на нем что-то пилить

**Comment by Znack** [comment]
Ну может быть, согласен, что твой список солидно уже выглядит :)
